cf/post-register-lambda-v2.yaml

```diff
@@ -0,0 +1,229 @@
+
+Outputs:
+  PostRegistrationLambdaArnV2:
+    Description: ARN of PostRegistrationLambda
+    Value: !GetAtt PostRegistrationLambda.Arn
+    Export:
+      Name: !Sub PostRegistrationLambdaArnV2-${Environment}
+
+Conditions:
+  isProd: !Equals [!Ref Environment, "production"]
+
+Resources:
+  PostRegistrationLambdaRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: "sts:AssumeRole"
+      Path: /
+      Policies:
+        - PolicyName: !Sub "can-access-dynamo-lambda-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "dynamodb:DeleteItem"
+                  - "dynamodb:GetItem"
+                  - "dynamodb:PutItem"
+                  - "dynamodb:Scan"
+                  - "dynamodb:UpdateItem"
+                Resource:
+                  - Fn::Sub: "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/user-access-${Environment}*"
+                  - Fn::Sub: "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/invite-access-${Environment}*"
+        - PolicyName: !Sub "can-create-log-group-in-cloudwatch-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "logs:CreateLogGroup"
+                Resource:
+                  - Fn::Sub: "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:*"
+        - PolicyName: !Sub "can-add-logs-in-cloudwatch-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "logs:CreateLogStream"
+                  - "logs:PutLogEvents"
+                Resource:
+                  - Fn::Sub: "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/post-register-invite-lambda-${Environment}:*"
+        - PolicyName: !Sub "can-post-to-post-registration-sns-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - 'sns:Publish'
+                Resource:
+                  - Fn::Sub: "arn:aws:sns:${AWS::Region}:${AWS::AccountId}:post-registration-user-access-${Environment}-default"
+
+  PostRegistrationTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      KmsMasterKeyId: "alias/aws/sns"
+      TopicName: !Sub "post-registration-user-access-${Environment}-default"
+
+  PostUserRegistrationSubscription:
+    Type: 'AWS::SNS::Subscription'
+    Properties:
+      TopicArn: !Ref PostRegistrationTopic
+      Endpoint: !If [isProd, "https://api.scp.biomage.net/v2/access/post-registration",
```

Comment:
Is there a way to get the api endpoint dynamically without manually inputting them?

Reply:
Right now no. We havea lot of places where we hard code the endpoint. it would be nice to do this sometime in the future though, by storing it in config file or somewhere in AWS.

Reply:
Yes, this is breaking the multi-account deployments. Some time in the future is now btw, but feel free to leave it like this for this ticket and we will fix this and all the hardcodings after this is merged. We need to stop introducing new hardcodings though.